kube_downscaler/scaler.py

```diff
@@ -279,6 +303,18 @@ def autoscale_resource(
                             uptime,
                             downtime,
                         )
+                    elif resource.kind == "HorizontalPodAutoscaler":
+                        resource.obj["spec"]["minReplicas"] = target_replicas
```

Comment:
only modifying `minReplicas` will not really have an effect, should we change both min and max?

Reply:
when i mentioned the following in README 
```
An additional use-case with HPA is where you adjust the minReplicas of the HPA during downtime, using the kube-downscaler annotations, thus allowing the deployments to be scaled independently based on the HPA metrics during the downtime. In such a case, apply the kube-downscaler annotations only to the HPA.
```

What I meant was if someone wants to downscale their minimum replicas during the "dowmtime" but still have the ability to scale based on HPA metrics, they would only have to add annotations to the HPA. This is useful in cases where the HPA minReplicas is high during normal functioning of the day due to high load anticipation, but the same minReplicas can be scaled down during the night-time(downtime).

The above is jsut a bonus use-case I realized while documenting the HPA feature.

As far avoiding HPA and kube-downscaler race conditions, the conditions in the code should suffice.

Let me know your thoughts.

Reply:
I am not realizing what you meant. If the maxReplicas count is left untouched and both the HPA abe deployment have the annotations, out can lead to trace conditions of the HPA devices to scale.


Turns out my code changes only work for the "bonus case" where you only reduce the minReplicas of the HPA without adding any annotations to the deployment.

I think that aisi suffices the downscaling needs. Of course, adjusting both the min and max replicas of the HPA will be more generic.

Reply:
I have been approaching this problem differently especially while explaining it. I've modified the README file to provide clarity. I hope that resolves your concerns as far as how this feature should be used and if the code changes are enough.